activemodel/test/cases/validations/length_validation_test.rb

```diff
@@ -124,6 +124,29 @@ def test_validates_length_of_using_within_with_exclusive_range
     assert_predicate t, :valid?
   end
 
+  def test_validates_length_of_using_within_with_infinite_ranges
+    [
+      [-Float::INFINITY..10,                "a" * 10,   "a" * 11],
+      [-Float::INFINITY...10,               "a" * 9,    "a" * 10],
+      [-Float::INFINITY..,                  "a" * 10,   nil],
+      [10..Float::INFINITY,                 "a" * 10,   "a" * 9],
+      [10...Float::INFINITY,                "a" * 10,   "a" * 9],
+      [10..,                                "a" * 10,   "a" * 9],
+      [-Float::INFINITY..Float::INFINITY,   "a" * 10,   nil],
+    ].each do |range, valid, invalid|
```

Comment:
What about beginless ranges (e.g. `..10`)?  We should cover those too, yes?

Also, what do you think about always using the same valid title?  i.e.  Always use `"a" * 10`, but change `-Float::INFINITY...10` to `-Float::INFINITY...11`.  We could also extract the `"a"`s, so this becomes `each do |range, invalid_length|`.  That might make this code a bit easier to digest.

Reply:
I was sure I missed a case or two.
Thanks, much better now. Updated.